examples/websocket/websocket_example_app.html

```diff
@@ -0,0 +1,266 @@
+             font-weight: bold;
+             margin-left: 10px;
+             color: var(--header-color);
+             transition: color 0.3s;
+         }
+
+         .message {
+             background-color: var(--message-bg);
+             border: 1px solid var(--message-border);
+             border-radius: 4px;
+             padding: 15px;
+             margin-bottom: 15px;
+             transition: background-color 0.3s, border-color 0.3s;
+         }
+
+         .message:nth-child(even) {
+             background-color: var(--message-alt-bg);
+         }
+
+         .timestamp {
+             font-size: 0.8em;
+             color: var(--timestamp-color);
+             transition: color 0.3s;
+         }
+
+         .error {
+             color: var(--error-color);
+             margin-top: 5px;
+             transition: color 0.3s;
+         }
+
+         @media (max-width: 600px) {
+             body {
+                 padding: 10px;
+             }
+
+             .container {
+                 padding: 15px;
+             }
+
+             input[type="text"], #json-input, button {
+                 font-size: 14px;
+             }
+         }
+     </style>
+ </head>
+ <body>
+     <div class="container">
+         <h1>WebSocket Example App</h1>
+         <p>This is a simple app to show how JSON can be sent into a solace-ai-connector flow and how to receive output from it. Just hit connect to connect to the flow, type in some JSON and hit send. Your JSON should be echoed back to you.</p>
+
+         <div id="connection-section">
+             <input type="text" id="url-input" value="ws://localhost:5000">
+             <button id="connect-button">Connect</button>
+             <span id="status">Disconnected</span>
+         </div>
+
+         <div id="input-section">
+             <textarea id="json-input" placeholder="Enter JSON here"></textarea>
+             <button id="send-button">Send</button>
+             <div id="input-error" class="error"></div>
+         </div>
+
+         <div id="output-section">
+             <h2>Received Messages</h2>
+             <div id="messages"></div>
+             <button id="clear-button">Clear Messages</button>
+         </div>
+     </div>
+
+     <script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.0.1/socket.io.js"></script>
+     <script>
+         const urlInput = document.getElementById('url-input');
+         const connectButton = document.getElementById('connect-button');
+         const status = document.getElementById('status');
+         const jsonInput = document.getElementById('json-input');
+         const sendButton = document.getElementById('send-button');
+         const inputError = document.getElementById('input-error');
+         const messagesDiv = document.getElementById('messages');
+         const clearButton = document.getElementById('clear-button');
+
+         let socket;
+
+         function connect() {
+             socket = io(urlInput.value);
```

Comment:
Aren't the input and output endpoints on 2 different servers?

Reply:
No. By default, the websocket_output will look into the global pool of open websocket connections using the socket_id from its input. It is possible, but not typical, for the websocket_output to have a server, but that should only be for cases where it isn't a request/response type of application.